src/vs/platform/clipboard/browser/clipboardService.ts

```diff
@@ -4,15 +4,83 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { $ } from 'vs/base/browser/dom';
+import Severity from 'vs/base/common/severity';
 import { URI } from 'vs/base/common/uri';
+import { localize } from 'vs/nls';
 import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+import { ILogService } from 'vs/platform/log/common/log';
+import { IOpenerService } from 'vs/platform/opener/common/opener';
 
 export class BrowserClipboardService implements IClipboardService {
 
+	constructor(
+		@IDialogService private readonly dialogService: IDialogService,
+		@IOpenerService private readonly openerService: IOpenerService,
+		@ILogService private readonly logService: ILogService,
+		@IEnvironmentService private readonly environmentService: IEnvironmentService,
+	) { }
+
 	declare readonly _serviceBrand: undefined;
 
 	private readonly mapTextToType = new Map<string, string>(); // unsupported in web (only in-memory)
 
+	// Guard access to navigator.clipboard with try/catch
+	// as we have seen DOMExceptions in certain browsers
+	// due to security policies. For example, in Safari,
```

Comment:
Are we sure this works in Safari as we intended? Previously in our clipboard service we had this note:

https://github.com/microsoft/vscode/blob/e4373f7591bbbc96ed10007bf0f22d2f6e2006e9/src/vs/workbench/services/clipboard/browser/clipboardService.ts#L41-L45